File: 算法/排序/冒泡排序.php
<?php

/**
 * @intro
 * User: HuangHaoyu
 * Email: haoyuhuang@sfmail.sf-express.com
 * Date: 2021/12/14
 * Time: 6:45 下午
 */
class BubbleSort {
    public function sort($nums) {
        if(empty($nums) || count($nums) == 1) {
            return $nums;
        }

        $length = count($nums) - 1;
        for($i = 0; $i <= $length; $i ++) {
            for($j = 1; $j <= $length - $i; $j++) {
                if($nums[$j] < $nums[$j - 1]) {
                    list($nums[$j - 1], $nums[$j]) = [$nums[$j], $nums[$j - 1]];
                }
            }
        }

        return $nums;
    }
}

$bubbleSort = new BubbleSort();
print_r($bubbleSort->sort([5,1,1,2,0,0]));

File: 算法/排序/冒泡排序.go
package 排序

func BubbleSort(arr []int) {
	if len(arr) <= 1 {
		return
	}

	l := len(arr)
	for i := 0; i < l; i++ {
		for j := 1; j < l; j++ {
			if arr[j] < arr[j-1] {
				arr[j], arr[j-1] = arr[j-1], arr[j]
			}
		}
	}

	return
}
